Donation-platform/src/main/java/com/google/DbCalls.java

```diff
@@ -110,17 +97,144 @@ public void addUser(String userName, String email,
+  public Charity getCharityById(Key charityId) throws Exception{
+    Entity charityEntity = datastore.get(charityId);
+    Charity charityClass = setCharityClass(charityEntity);
+    return charityClass;
+  }
+  // Function gets one charity by name. Returns error if multipe charities
+  // have the same name.
+  public Charity getCharityByName(String name) throws Exception {
+    Query query =
+    new Query("Charity")
+        .setFilter(new FilterPredicate("name", FilterOperator.EQUAL, name));
+    PreparedQuery pq = datastore.prepare(query);
+    Entity result = pq.asSingleEntity();
+    Charity charityResult = setCharityClass(result);
+    return charityResult;
+  }
+  // Function returns all charities under a category tag.
+  public Collection<Charity> getCharitiesByTag(String name) throws Exception {
+    ArrayList< Charity > charityDataStore = new ArrayList < Charity > ();
+    Query query = new Query("Charity");
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable())
+        {
+            Charity charityData = setCharityClass(entity);
+            // Read the categories assigned to the charity
+            Collection<Key> categoryKeys = charityData.getCategories();
+            if (categoryKeys != null){
+              for (Key key : categoryKeys ){
+                Entity tagEntity = datastore.get(key);
+                Tag tagClass = setTagClass(tagEntity);
+                String tagName = tagClass.getName();
+                // Check to see if category contains tag searched for.
+                if (tagName.equals(name)){
+                  charityDataStore.add(charityData);
+                }
+              } 
+            } 
+    }
+    // Return all charity objects containing tag.
+    return charityDataStore;
+  }
+  // Function returns tag object from ID.
+  public Tag getTagById(Key tagId) throws Exception{
+    Entity tagEntity = datastore.get(tagId);
+    Tag tagClass = setTagClass(tagEntity);
+    return tagClass;
   }
-  // Function updates trending score of a tag in the database.
-  public void updateTagTrendingScore(Key id, Double trendingScore) throws Exception{
+  // Function returns tag object matching name passed in.
+  public Tag getTagByName(String name) throws Exception{
+    Query query =
+    new Query("Tag")
+        .setFilter(new FilterPredicate("name", FilterOperator.EQUAL, name));
+    PreparedQuery pq = datastore.prepare(query);
+    Entity result = pq.asSingleEntity();
+    Tag tagResult = setTagClass(result);
+    return tagResult;
+  }
+  //Function that returns user from user name.
+  public Users getUserByUserName(String userName) throws Exception{
+    Query query =
+    new Query("Users")
+        .setFilter(new FilterPredicate("userName", FilterOperator.EQUAL, userName));
+    PreparedQuery pq = datastore.prepare(query);
+    Entity result = pq.asSingleEntity();
+    Users userResult = setUsersClass(result);
+    return userResult;
```

Comment:
You can write this as: 

return setUsersClass (datastore.prepare(query). asSingleEntity());